File: apps/api/app/core/tts.py
from dataclasses import dataclass


@dataclass(frozen=True)
class TTSVoiceConfig:
    language: str
    voice_name: str
    display_name: str


MINIMAX_TTS_VOICES: dict[str, TTSVoiceConfig] = {
    "zh": TTSVoiceConfig(
        language="zh",
        voice_name="minimax_zh_female",
        display_name="MiniMax 中文女声",
    ),
    "en": TTSVoiceConfig(
        language="en",
        voice_name="minimax_en_female",
        display_name="MiniMax English Female",
    ),
    # TODO: Replace fallback voice_name values with provider-confirmed multilingual voices.
    "ja": TTSVoiceConfig(
        language="ja",
        voice_name="minimax_en_female",
        display_name="MiniMax Japanese fallback",
    ),
    "ko": TTSVoiceConfig(
        language="ko",
        voice_name="minimax_en_female",
        display_name="MiniMax Korean fallback",
    ),
    "es": TTSVoiceConfig(
        language="es",
        voice_name="minimax_en_female",
        display_name="MiniMax Spanish fallback",
    ),
    "fr": TTSVoiceConfig(
        language="fr",
        voice_name="minimax_en_female",
        display_name="MiniMax French fallback",
    ),
    "ru": TTSVoiceConfig(
        language="ru",
        voice_name="minimax_en_female",
        display_name="MiniMax Russian fallback",
    ),
}


def get_tts_voice(language: str) -> TTSVoiceConfig:
    return MINIMAX_TTS_VOICES.get(language, MINIMAX_TTS_VOICES["en"])

File: apps/web/lib/tts.ts
import type { Locale } from "@/components/LanguageProvider";

export const ttsVoiceLabels: Record<string, string> = {
  minimax_zh_female: "Kaiqiang 中文女声",
  minimax_en_female: "Kaiqiang English Female",
  BV001_streaming: "Kaiqiang 女声",
  BV002_streaming: "Kaiqiang 男声",
};

// TODO: Replace fallback voice_type values with provider-confirmed multilingual voices.
export const defaultVoiceTypeByLocale: Record<Locale, string> = {
  zh: "BV001_streaming",
  en: "BV001_streaming",
  ja: "BV001_streaming",
  ko: "BV001_streaming",
  es: "BV001_streaming",
  fr: "BV001_streaming",
  ru: "BV001_streaming",
};

export function getDefaultVoiceTypeForLocale(locale: Locale) {
  return defaultVoiceTypeByLocale[locale] ?? defaultVoiceTypeByLocale.en;
}

export function getTTSVoiceLabel(voiceName: string) {
  return ttsVoiceLabels[voiceName] ?? voiceName;
}

export function getLanguageLabel(language: string) {
  if (language === "zh") return "中文";
  if (language === "en") return "English";
  if (language === "ja") return "日本語";
  if (language === "ko") return "한국어";
  if (language === "es") return "Español";
  if (language === "fr") return "Français";
  if (language === "ru") return "Русский";
  return language;
}
